src/lakefs_spec/spec.py

```diff
@@ -421,15 +441,37 @@ def ls(self, path: str | os.PathLike[str], detail: bool = True, **kwargs: Any) -
         # cache the info if not empty.
         if info:
             # assumes that the returned info is name-sorted.
+            # All directories included in this listing
+            directories = {str(Path(e["name"]).parent) for e in info}
+
             pp = self._parent(info[0]["name"])
```

Comment:
These two should be consistent to yield the same results, so either change to `self._parent(e["name"])` above or to the `Path().parent` approach below.

Reply:
Actually, the code line just above was a leftover from debugging; removed it.